### DISCUSSION TITLE
Request a firmware build - 2.7.1

### DISCUSSION THREAD

**Discussion by pfeerick** [EdgeTX/edgetx#1951] [General]
Ask here if you want/need specific options that are not included in the stock builds. Builds will be uploaded to [here (same link as 'By request' below)](http://files.portabledev.xyz/d/f83d44a185ea4895ac97/), so check it out before making a request just in case there is already a custom build with your chosen options...

*Please remember to specify your transmitter when requesting a build* ;)

- [Multilingual builds](http://files.portabledev.xyz/d/d87fe9040f844253b2b4/) - zip file per transmitter containing all languages built
- [Multilingual builds - NOHELI](http://files.portabledev.xyz/d/7e7a668d887d4930a565/) - zip file per transmitter containing all languages built
- [By request](http://files.portabledev.xyz/d/f83d44a185ea4895ac97/)

**Comment by chewi1472** [comment]
Hello pfeerick
May i have a built FR for TX16S with no Heli and bluetooth.
Thank you, and have a good day

**Reply by ArdSchell** [reply]
That would be very nice indeed.
TX16S with bluetooth enabled.
¡ǝɔuɐʌpɐ uı sʞuɐɥʇ


TX16S_EN_BLUETOOTH

**Reply by Xanthuz168** [reply]
hi, TX16S_En_Bluetooth please... thank you so much..

**Reply by pfeerick** [reply]
@chewi1472 Apologies for the delay. Available in the specials link in the first post.

```cmake -Wno-dev -DPCB=X10 -DPCBREV=TX16S -DBLUETOOTH=Y -DHELI=N -DTRANSLATIONS=FR .. ```

**Reply by pfeerick** [reply]
@ArdSchell @Xanthuz168 Apologies for the delay. Available in the specials link in the first post, both with and without heli tab.

```
cmake -Wno-dev -DPCB=X10 -DPCBREV=TX16S -DBLUETOOTH=Y ..
cmake -Wno-dev -DPCB=X10 -DPCBREV=TX16S -DBLUETOOTH=Y -DHELI=N ..
```

**Comment by ArdSchell** [comment]
Hello Pfeerick,

Would be great if you could make a TX16S with bluetooth enabled. 
TX16S_EN_BLUETOOTH

Thanks in advance!

Best regards,

Ard

**Reply by spiky1** [reply]
Hallo Pfeerick

I would like a 2.7.1 multilingual noheli build for the X12s with access.

**Reply by spiky1** [reply]
Hallo Pfeerick,

I should add to my request above:
X12S with internal Access multilingual or DE

Thanks and best regards, Hans

**Reply by pfeerick** [reply]
@spiky1 You should be able to use the multi-lingual noheli builds above - just pick the DE firmware version since that's what you are after. And once you've loaded the firmware up, set the internal RF type to ISRM (instead of XJT for ACCST). 

![image](https://user-images.githubusercontent.com/5500713/171067382-9078e0d8-6009-4410-9878-99d3432902fe.png)

**Reply by spiky1** [reply]
Hallo Pfeerick,

I tried that before I asked.
Today I tried again but I can only choose between off and XJT, no other choice available.
![image](https://user-images.githubusercontent.com/50096813/171357610-529313a4-aa45-4b46-9c2c-702753e5498d.jpeg)

Firmware loaded is edge-tx x12s a55aff02 with DE.

Also tried 2.7.0 f79978b6e with the same result.

**Reply by spiky1** [reply]
@freerick With opentx 2.3.15 internal access I can select the ISRM option.

**Reply by pfeerick** [reply]
You need to go to Radio Settings -> Hardware -> Internal RF and change that to ISRM (per my earlier screenshot). *Then* your model settings page internal RF list will give OFF and ISRM as options ;)

**Reply by spiky1** [reply]
@pfeerick some people are slow learners🤷
I found the right place and it now works! 🥂

Thx for your help (and patience!)

**Reply by pfeerick** [reply]
I wouldn't go quite *that* far... it is a new option (was introduced in ETX 2.6)... and language barrier can lead to some interesting times also ;) Glad it is all working for you now. Enjoy! :)

**Comment by ETHOSWarrier** [comment]
Hello,I have a Frsky Horus x10 that i have installed the ISRM  RF module in  to allow it to run ACCESS firmware but there is no EdgeTx build for it...I found out the hard way when trying the "ACCESS" labelled download but got the wrong PCB error!...

My build would be a standard build like the ones downloadable already....Thanks in Advance...Chris.

**Reply by ETHOSWarrier** [reply]
Sorry forgot to say EU LBT would be needed for ACCST D16 mode  and in English language...cheers.

**Reply by chewi1472** [reply]
Hello,
 with a standard firmware,
switch on your radio
go to: SYS-->HARDWARE
at the end of the menu -->internal modul: select ISRM.
Go to your model ans select in internal modul what your want: ACCESS or D16.

**Reply by ETHOSWarrier** [reply]
Thats the issue,The standard firmware bricks the radio and you only get the "Wrong PCB Error"...you cannot start the radio to change anything.This has happened to others so a special build for the IRSM converted x10 is required.

**Reply by pfeerick** [reply]
So you are saying that if you load the standard firmware , it will brick the radio? As if your radio is a X10 with exchanged RF board, it's still a X10 mainboard, not an X10 Access one. You would then set the internal RF type to ISRM.

**Reply by alant101** [reply]
Following this with great interest as I have the same transmitter with the IRSM RF and would like to resolve this problem

**Reply by pfeerick** [reply]
So it's a X10/X10S ACCST mainboard with ISRM RF board? Try the X10 firmware (*NOT* the X10-ACCESS which is for the Express radios), and then Radio Settings -> Hardware -> Internal RF to set the internal RF to ISRM.

**Comment by arfpv** [comment]
Has anyone used the TBS Crossfire nano with the Radio master zorro?
I cant seem to run the Lua script. Updated openTX to 2.7.1, still get the same syntax error.

**Comment by MadMonkey87** [comment]
Hello Pfeerick

I would like a 2.7.1 build for the TX16s with noheli but with LUA_MIXER

Thank you in advance!

**Reply by pfeerick** [reply]
@MadMonkey87 Sorry for the delay. Available in the specials link in the first post. 
```
cmake -Wno-dev -DPCB=X10 -DPCBREV=TX16S -DHELI=N -DLUA_MIXER=Y ..
```

**Reply by gjhaynes** [reply]
Thanks Peter. Brilliant!
Do you have a timeline for inclusion of Custom Scripts in a production release?

**Reply by pfeerick** [reply]
At this point no. I know it has never been liked as a enabled by default option as it means you have a dependence of there being zero bugs with the script, and for Lua to not crash... However, we are probably getting to the stage now where the core of it has been used enough and people can be responsible for their own decisions ;)

**Reply by gjhaynes** [reply]
Thanks Peter. that makes sense. I would be OK with the EdgeTx Team issuing a disclaimer about consequences of using Custom Scripts. It would be really cool to allow user editing of the Model menu tabs so they only see what is useful to them, e.g. hide Heli and Custom Scripts. Of course, this means a universal build that provides for every option and this may not be an efficient use of hardware resources. Maybe you could check user prefs during startup and load only the relevant optional modules. But I'm sure this is not a trivial task!

**Reply by pfeerick** [reply]
True. But this is exactly the direction we want to go... hence https://github.com/EdgeTX/edgetx/issues/1692 However I don't know if this will make 2.8 now as I haven't even started on it and no-one else has raised their hand to work on it yet. :grin: With colorlcd radios having enough space for the firmware to have pretty much all the options enabled, there's no reason not to go that way. For B&W radios, though, it will have to stay as it is as there is simply not enough space.

**Reply by gjhaynes** [reply]
I'm so impressed with your responsiveness! Even more impressive given the workload you guys must be currently handling.
Long Live EdgeTx !!

**Reply by gjhaynes** [reply]
Hi Peter. Is there a downloadable 2.8 Companion & Simulator to play with?
Is this something I can create myself with GitPod?

**Reply by pfeerick** [reply]
Check the github actions pages for the Windows/Mac/Linux builds for the latest companion builds for the main branch. You *could* build a linux version on GitPod, but that won't help you if on Windows. Basically each build needs to run on the OS it is targeted for - Windows on Windows, Mac on Mac, Linux on Linux. 

https://github.com/EdgeTX/edgetx/actions/workflows/win_cpn-64.yml?query=branch%3Amain
https://github.com/EdgeTX/edgetx/actions/workflows/macosx_cpn.yml?query=branch%3Amain
https://github.com/EdgeTX/edgetx/actions/workflows/linux_cpn.yml?query=branch%3Amain

**Comment by WizzX-FPV** [comment]
Hallo, 
I would like to ask for 2.7.1 build for TX16S with:
ppmus
noheli
bluetooth
lua

when can we expect user friendly options selection?
Thank you.

**Reply by pfeerick** [reply]
@WizzX-FPV Sorry for the delay. Available in the specials link in the first post.

Please let me know if the `ppmus` option is working properly... I don't know how much testing option has had in the new UI as this is the first build I remember doing for a colorlcd target. 

I don't have any idea of the timeline atm, I've been out of the loop for last two weeks but there hadn't been any progress of late. RL getting in the way, etc. Wasn't sure if you wanted lua or the lua custom mixer scripts, so I did both. There has been a misconception (due to how the option was presented) for a long time the LUA checkbox meant support for Lua when it fact it meant enabling Lua custom mixer scripts.

```
cmake -Wno-dev -DPCB=X10 -DPCBREV=TX16S -DPPM_UNIT=US -DHELI=N -DBLUETOOTH=Y ..
cmake -Wno-dev -DPCB=X10 -DPCBREV=TX16S -DPPM_UNIT=US -DHELI=N -DBLUETOOTH=Y -DLUA_MIXER=Y ..
```

**Comment by ArdSchell** [comment]
Hello Pfeerick,

Is the TX16S_EN_BLUETOOTH already available? or am I missing something.
Thanks you for your great help.

Best regards,

Ard

**Reply by pfeerick** [reply]
No, you're not missing anything... more I've been missing for a while ;) It's up in the special link in first post.

**Reply by ArdSchell** [reply]
Thanks Pfeerick,

Hope you are oke and welcome back.
Thanks for sharing the files.
:O)

**Reply by ArdSchell** [reply]
I updated my TX16S to 2.7.1 and it works perfect. 
Thanks again Pfeerick.
¡ʇsǝʇɐǝɹƃ ǝɥʇ ɹnoʎ

**Comment by Meldinov** [comment]
Hi, i habe a problem with the 2.7.1 t-pro HELI. both ES and EN wont boot, the noheli works great.

**Reply by pfeerick** [reply]
I just checked the HELI EN and ES firmwares from the tpro HELI zip on my own T-Pro (MPM variant) and had no issues with them booting when flashed from the bootloader. So not sure what it's going on there.

**Comment by tyxiaotx** [comment]
Hello pfeerick,
May I have a build for Jumper T16 with bluetooth enabled? Thank you!

**Reply by pfeerick** [reply]
Done... check the specials link in the first post. 
`cmake -Wno-dev -DPCB=X10 -DPCBREV=T16 -DINTERNAL_MODULE_MULTI=ON -DBLUETOOTH=ON ..`
`cmake -Wno-dev -DPCB=X10 -DPCBREV=T16 -DINTERNAL_MODULE_MULTI=ON -DBLUETOOTH=ON -DHELI=N ..`

**Comment by koakumaping** [comment]
Hello pfeerick,
May I have a build for xlite pro with ACCST D16 D8(i dont use ACCESS), no heli, EN,   ver 2.7.1, thx

**Reply by pfeerick** [reply]
You should be able to use the standard EN_NOHELI build... just set your internal RF to XJT instead of ISRM. 

https://user-images.githubusercontent.com/5500713/177679821-b28f9ac4-68b8-47ad-a556-23b7d61e54b5.mp4

**Reply by koakumaping** [reply]
ok,i will try it later

**Comment by Billy60** [comment]
Hello all
The links to the Multilinguals are dead.
Could they be fixed ?
Thanks a lot !

**Reply by pfeerick** [reply]
Looks like the VPS had an outage yesterday and didn't come back up again properly... should be right now. Sorry about that! :)

**Comment by shihuntaotie** [comment]
Hello,
May i have a built FR for X9DP with chinese?
Thank you

**Reply by pfeerick** [reply]
Unfortunately, I don't think so - I don't think CN or TW are working on B&W radios at the moment. I'll double-check, but I'm pretty sure you get a whole lot of nothing where there should be Chinese characters... which isn't very useful! :-O

**Comment by Billy60** [comment]
My tx16s is indeed colour
Too bad
Thanks for trying !

Le dim. 11 sept. 2022, 07:22, Peter Feerick ***@***.***> a
écrit :

**Reply by pfeerick** [reply]
Um, you might want to look again - I wasn't replying to you - and I'm not sure how I'd know what radio you have since you never said! 😆 That was a reply to "X9DP with chinese" which is a B&W radio ;) I replied to you 2-3 days ago letting you know the links should be working again ;) "Looks like the VPS had an outage yesterday and didn't come back up again properly... should be right now. Sorry about that! :)"

**Comment by pfeerick** [comment]
Build for X7 with NOHELI and LUA_MIXER added
`cmake -Wno-dev -DHELI=N -DLUA_MIXER=Y -DPCB=X7 ..`

**Comment by 2xWhiskey** [comment]
Hello, is there a specific build option to utilize CRSF (Crossfire Shot) internally with the radio's internal RF module?  I've selected CRSF and 400k Baud rate under the radio's HARDWARE menu as well as powered the internal module via the model specific menu.  I'm using a Taranis X-Lite Pro (EdgeTX 2.7.1) with a DIY ELRS TX module (2.5.1) and have been unable to get the module to communicate via the Lua script.  I have verified the module is working by connecting it to the radio's external bay and successfully using the Lua script.

**Reply by pfeerick** [reply]
This is more a general or Q&A type question than a firmware build request one. Especially since you need to consider aspects such as "how is the RF module connected to the radio" - i.e. does it have a full duplex connection? #2337 Are you using that connection, as defined in the [`hal.h`](https://github.com/EdgeTX/edgetx/blob/main/radio/src/targets/taranis/hal.h)? From a quick look at that, that transmitter should work as is, but I've not heard of it *actually* done before, just talked about, so you could be entering uncharted waters, and settings may still need to be tweaked ;)

**Reply by 2xWhiskey** [reply]
Thank you, I am using a copycat of the TX_SX1280_Minimal design.  It utilizes one pin connection for communications, Pin 10 (SPORT), of the main 15 pin connector that goes from the main board to the internal RF board.  Does that SPORT line count as full duplex?  Or are you referring to something like a UART that has both an RX and TX line?  I tried looking at the hal.h you linked, but honestly I don't understand how to interpret it.  I did reconnect the module to the external bay connect and verified that it was communicating with the ELRS lua script.

**Reply by 2xWhiskey** [reply]
I was just thinking, I don't know the official pinout orders, just what I wrote down from taking notes with a multimeter.  The SPORT line could also be Pin 5.

**Reply by pfeerick** [reply]
Hm, that could be a problem... full duplex means two-way communication at the same time, which for digital basically necessitates separate TX and RX. Half-duplex means only one side can talk at a time, so one line can be used for both. I have to admit I don't fully know how these are configured, but I believe the S.PORT line has the UART TX and RX both connected in some fashion. Another thing you may need to play with is the `UART_INVERTED` setting - that may have some bearing now that you are not going though the external sport connection. @raphaelcoeffic will be able to give a more informed overview of what's needed. Maybe also ask over on the ELRS discord, the `#handset` channel is probably a good place to start.

**Reply by 2xWhiskey** [reply]
Thank you for your advice and time.  I do appreciate it!

**Comment by federikk** [comment]
Hello @pfeerick ,
may i ask a 2.7.1 for RM tx16s with:
crossfire, ghost, flexr9m, afhds3, internalmulti, multimodule, lua, luac, externalaccessmod

multilingual (IT)
thanks a lot
F

**Reply by pfeerick** [reply]
Sorry for the delay... flexr9m is no longer needed as an option as flex is enabled by default now, so the only extra options needed are for the external access mod and language. Check the "By Request" link at the top of the page for the firmware. 

```
cmake -Wno-dev -DPCB=X10 -DPCBREV=TX16S -DTRANSLATIONS=IT -DHARDWARE_EXTERNAL_ACCESS_MOD=Y ..
```

**Comment by azaz44** [comment]
**Update**: I was able to compile it on my own. Although I'm not sure if I got the flags right. Would it be true to say, that such setup:
```
cmake -Wno-dev -DPCB=sth -DPCBREV=sth
```
done on 2.7, would give output equal to official release Black Pearl?
I had troubles to find out how is actually official release compiled.

---------------


Hi. Could I request x10 (non-access) with:
- noheli
- LUA_MIXER=Y
- others like in official release of Black Pearl

BTW, is there any reason mixing scripts are not included in official builds? Is it not stable?

**Reply by pfeerick** [reply]
`tools/build-gh.sh` is used to generate all the firmwares, so to generate the same as 2.7.1, if building from that tag on GH, the `cmake` used to generate the firmware for the non-access X10 would look like 

```
cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_RULE_MESSAGES=OFF -Wno-dev -DPCB=X10 ..
```

but for all intents and purposes, this is not functionally different from what you gave as most of that extra is just for hiding cmake messages we're not interested in seeing during the build process. 

i.e. for what you wanted, I would have used 
```
cmake -Wno-dev -DPCB=X10 -DLUA_MIXER=Y -DHELI=N ..
```

**Reply by azaz44** [reply]
Thx, this is great help 👍

**Reply by Helipro** [reply]
I am looking to get 2 compiled version of 2.9 for my Bluetooth build T16s and Jumper Tx16.  I was not able to compile my own as I was with previous versions.
cmake -Wno-dev -DPCB=X10 -DPCBREV=T16 -DDEFAULT_MODE=2 -DPPM_UNIT=US -DBLUETOOTH=YES -DLUA=ON -DOVERRIDE_CHANNEL_FUNCTION=OFF   -DCMAKE_BUILD_TYPE=Release ../

cmake -Wno-dev -DPCB=X10 -DPCBREV=TX16S -DDEFAULT_MODE=2 -DPPM_UNIT=US -DBLUETOOTH=YES -DLUA=ON -DOVERRIDE_CHANNEL_FUNCTION=OFF   -DCMAKE_BUILD_TYPE=Release ../

Thanks,
Craig

**Reply by pfeerick** [reply]
As in you don't have the facility to do so, or encountered issues in building them? Both those build options built fine for me. 

[firmwares.zip](https://github.com/EdgeTX/edgetx/files/12508728/firmwares.zip)

**Reply by Helipro** [reply]
Thank you I really appreciate it.  I encountered issues:  I tried the same way with past versions.  Error that I receive now:
CMake Error: The source directory "/workspace" does not appear to contain CMakeLists.txt.
Config error: make: *** No rule to make target 'configure'.  Stop.
![1](https://github.com/EdgeTX/edgetx/assets/119552818/4fc0a0b3-1327-4c14-87db-8e60c9c4aa3c)

**Reply by pfeerick** [reply]
Ok, you're hitting a problem we've noticed with the v2.9.0 version tag and gitpod... it's not executing the initialisation task whereby it would run `mkdir build && cd build`, so since you're not in the build directory since cmake has been told to look in the next directory up (the `..` at the end) it can't find it. It seems even it you do create that build directory that gitpod doesn't seem to be initialising properly as it doesn't seem to be able to find the compiler. However, if you instead checkout the 2.9 branch as it is it seems to be working properly... i..e. https://gitpod.io/#https://github.com/edgetx/edgetx/tree/2.9

**Reply by Helipro** [reply]
The 2 files you sent me for the T16 and Tx16s bricked both transmitters in the middle of the update.
I am also still not able to access the web link you sent me to build my own firmware.  Could you try sending me the firmware again, Thanks Craig
![2](https://github.com/EdgeTX/edgetx/assets/119552818/f0c09b75-81aa-4a75-be83-51bed59e319a)

**Comment by pfeerick** [comment]
By "bricked" do you mean the UI pdate via bootloader froze when it was
nearly at the end? If so, that happens occasionally, and after a hard power
cycle it usually updates successfully. We're not sure why that happens yet.

**Reply by Helipro** [reply]
Yes that is what happened.  after it bricked I programed the Tx with the stmcube programmer and with the files you sent me but it was 2.9-rc1.  I just used stm32cube programmer and went back to 2.9-rc2 on both tx's.
